### ISSUE TITLE
Rabbitmq fails to verify public key

### ISSUE THREAD

**Issue by ovecka** [puphpet/puphpet#2320]
Due to change on rabbitmq.com side, puphpet + puppet are unable to verify GPG. 

When I vagrant up new machine with rabbitmq enabled, it fails after loading facts.
Any install with rabbitmq on ubuntu fails for me
```
vagrantfile:
    target: local
    vm:
        provider:
            local:
                box: puphpet/ubuntu1404-x64
                box_url: puphpet/ubuntu1404-x64
                box_version: '0'
                chosen_virtualizer: virtualbox
                virtualizers:
                    virtualbox:
                        modifyvm:
                            natdnshostresolver1: false
                        showgui: 0
                    vmware:
                        numvcpus: 1
                    parallels:
                        use_linked_clone: 0
                        check_guest_tools: 0
                        update_guest_tools: 0
                machines:
                    vflm_2vkf5b9nsk1v:
                        id: machine11
                        hostname: machine11.puphpet
                        network:
                            private_network: 192.168.56.101
                            forwarded_port:
                                vflmnfp_d1mpifd4tt7n:
                                    host: '5445'
                                    guest: '22'
                        memory: '512'
                        cpus: '1'
        provision:
            puppet:
                manifests_path: puphpet/puppet/manifests
                module_path: puphpet/puppet/modules
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
        synced_folder:
            vflsf_vi6hok6ysdbh:
                source: ./
                target: /var/www
                sync_type: default
                smb:
                    smb_host: ''
                    smb_username: ''
                    smb_password: ''
                    mount_options:
                        dir_mode: '0775'
                        file_mode: '0664'
                rsync:
                    args:
                        - '--verbose'
                        - '--archive'
                        - '-z'
                    exclude:
                        - .vagrant/
                        - .git/
                    auto: 'true'
                owner: www-data
                group: www-data
        usable_port_range:
            start: 10200
            stop: 10500
        post_up_message: ''
    ssh:
        host: 'false'
        port: 'false'
        private_key_path: 'false'
        username: vagrant
        guest_port: 'false'
        keep_alive: '1'
        forward_agent: 'false'
        forward_x11: 'false'
        shell: 'bash -l'
        insert_key: 'false'
    vagrant:
        host: detect
    proxy:
        http: ''
        https: ''
        ftp: ''
        no_proxy: ''
server:
    install: '1'
    packages: {  }
users_groups:
    install: '1'
    groups: {  }
    users: {  }
locale:
    install: '1'
    settings:
        default_locale: en_US.UTF-8
        locales:
            - en_GB.UTF-8
            - en_US.UTF-8
        timezone: UTC
firewall:
    install: '1'
    rules: {  }
cron:
    install: '1'
    jobs: {  }
nginx:
    install: '1'
    settings:
        default_vhost: 1
        proxy_buffer_size: 128k
        proxy_buffers: '4 256k'
        proxy_connect_timeout: 600s
        proxy_send_timeout: 600s
        proxy_read_timeout: 600s
        names_hash_bucket_size: 128
    upstreams: {  }
    vhosts:
        nxv_wqygooru38ym:
            server_name: awesome.dev
            server_aliases:
                - www.awesome.dev
            www_root: /var/www/awesome
            listen_port: '80'
            index_files:
                - index.html
                - index.htm
                - index.php
            client_max_body_size: 1m
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_port: '443'
            ssl_protocols: ''
            ssl_ciphers: ''
            rewrite_to_https: '1'
            spdy: '1'
            locations:
                nxvl_ek47xbkv6p1q:
                    location: /
                    try_files:
                        - $uri
                        - $uri/
                        - /index.php$is_args$args
                    fastcgi: ''
                    fastcgi_index: ''
                    fastcgi_split_path: ''
                    proxy: ''
                    proxy_redirect: ''
                nxvl_hybom9goqn64:
                    location: '~ \.php$'
                    try_files:
                        - $uri
                        - $uri/
                        - /index.php$is_args$args
                    fastcgi: '127.0.0.1:9000'
                    fastcgi_index: index.php
                    fastcgi_split_path: '^(.+\.php)(/.*)$'
                    fast_cgi_params_extra:
                        - 'SCRIPT_FILENAME $request_filename'
                        - 'APP_ENV dev'
                    set:
                        - '$path_info $fastcgi_path_info'
                    proxy: ''
                    proxy_redirect: ''
    proxies: {  }
apache:
    install: '0'
    settings:
        version: 2.4
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - proxy_fcgi
        - rewrite
    vhosts:
        av_0ez6apixfiwy:
            servername: awesome.dev
            serveraliases:
                - www.awesome.dev
            docroot: /var/www/awesome
            port: '80'
            setenv:
                - 'APP_ENV dev'
            setenvif:
                - 'Authorization "(.*)" HTTP_AUTHORIZATION=$1'
            custom_fragment: ''
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
            ssl_protocol: ''
            ssl_cipher: ''
            directories:
                avd_sqztbsshpwnt:
                    path: /var/www/awesome
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: ''
                    files_match:
                        avdfm_5k73r4wm6urr:
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: ''
                            provider: filesmatch
                    provider: directory
letsencrypt:
    install: '1'
    settings:
        email: ''
    domains: {  }
php:
    install: '1'
    settings:
        version: '56'
    modules:
        php:
            - cli
            - intl
        pear: {  }
        pecl: {  }
    ini:
        display_errors: 'On'
        error_reporting: '-1'
        session.save_path: /var/lib/php/session
        date.timezone: UTC
    fpm_ini:
        error_log: /var/log/php-fpm.log
    fpm_pools:
        phpfp_fttjvhm06gbo:
            ini:
                prefix: www
                listen: '127.0.0.1:9000'
                security.limit_extensions: .php
                user: www-user
                group: www-data
    composer: '1'
    composer_home: ''
xdebug:
    install: '0'
    settings:
        xdebug.default_enable: '1'
        xdebug.remote_autostart: '0'
        xdebug.remote_connect_back: '1'
        xdebug.remote_enable: '1'
        xdebug.remote_handler: dbgp
        xdebug.remote_port: '9000'
blackfire:
    install: '0'
    settings:
        server_id: ''
        server_token: ''
        agent:
            http_proxy: ''
            https_proxy: ''
            log_file: stderr
            log_level: '1'
        php:
            agent_timeout: '0.25'
            log_file: ''
            log_level: '1'
xhprof:
    install: '0'
wpcli:
    install: '0'
    version: v0.21.1
drush:
    install: '0'
    version: 8.0.5
ruby:
    install: '1'
    versions:
        rv_nt1s5hypjtx4:
            default: '1'
            bundler: '1'
            version: 1.9.3
            gems:
                - deep_merge
                - activesupport
                - vine
python:
    install: '1'
    packages: {  }
    versions: {  }
nodejs:
    install: '0'
    settings:
        version: '5'
    npm_packages: {  }
hhvm:
    install: '0'
    nightly: 0
    composer: '1'
    composer_home: ''
    settings: {  }
    server_ini:
        hhvm.server.host: 127.0.0.1
        hhvm.server.port: '9000'
        hhvm.log.use_log_file: '1'
        hhvm.log.file: /var/log/hhvm/error.log
    php_ini:
        display_errors: 'On'
        error_reporting: '-1'
        date.timezone: UTC
mariadb:
    install: '1'
    settings:
        version: '10.1'
        root_password: '123'
        override_options: {  }
    adminer: 0
    users:
        mariadbnu_94ody4qtj8vk:
            name: dbuser
            password: '123'
    databases:
        mariadbnd_wxkkkaikuovu:
            name: dbname
            sql: ''
    grants:
        mariadbng_49p1hxswyha4:
            user: dbuser
            table: '*.*'
            privileges:
                - ALL
mysql:
    install: '0'
    settings:
        version: '5.6'
        root_password: '123'
        override_options: {  }
    adminer: 0
    users:
        mysqlnu_5vn7vfik9azz:
            name: dbuser
            password: '123'
    databases:
        mysqlnd_3cm74z0mtk7z:
            name: dbname
            sql: ''
    grants:
        mysqlng_1vothr93wuf0:
            user: dbuser
            table: '*.*'
            privileges:
                - ALL
postgresql:
    install: '0'
    settings:
        global:
            encoding: UTF8
            version: '9.4'
        server:
            postgres_password: '123'
    databases: {  }
    users: {  }
    grants: {  }
    adminer: 0
mongodb:
    install: '0'
    settings:
        bind_ip: 127.0.0.1
        port: '27017'
    globals:
        version: 2.6.0
    databases: {  }
redis:
    install: '0'
    settings:
        port: '6379'
sqlite:
    install: '0'
    adminer: 0
    databases: {  }
mailhog:
    install: '0'
    settings:
        smtp_ip: 0.0.0.0
        smtp_port: 1025
        http_ip: 0.0.0.0
        http_port: '8025'
        path: /usr/local/bin/mailhog
beanstalkd:
    install: '0'
    settings:
        listenaddress: 0.0.0.0
        listenport: '11300'
        maxjobsize: '65535'
        maxconnections: '1024'
        binlogdir: /var/lib/beanstalkd/binlog
        binlogfsync: null
        binlogsize: '10485760'
    beanstalk_console: 0
rabbitmq:
    install: '1'
    settings:
        port: '5672'
    users: {  }
    vhosts: {  }
    plugins: {  }
elastic_search:
    install: '0'
    settings:
        version: 2.3.1
        java_install: true
    instances:
        esi_1x2z27w8v0c2:
            name: es-01
solr:
    install: '0'
    settings:
        version: 4.10.2
        port: '8984'
```
```
==> machine11: Info: Loading facts
==> machine11: Warning: Scope(Apt::Source[mariadb]): $required_packages is deprecated and will be removed in the next major release, please use package resources instead.
==> machine11: Warning: Scope(Apt::Source[nginx]): $key_source is deprecated and will be removed in the next major release, please use $key => { 'source' => http://nginx.org/keys/nginx_signing.key } instead.
==> machine11: Warning: Scope(Apt::Source[rabbitmq]): $include_src is deprecated and will be removed in the next major release, please use $include => { 'src' => false } instead
==> machine11: Warning: Scope(Apt::Source[rabbitmq]): $key_source is deprecated and will be removed in the next major release, please use $key => { 'source' => http://www.rabbitmq.com/rabbitmq-release-signing-key.asc } instead.
==> machine11: Warning: Scope(Apt::Key[Add key: 7BD9BF62 from Apt::Source nginx]): $key_source is deprecated and will be removed in the next major release. Please use $source instead.
==> machine11: Warning: Scope(Apt::Key[Add key: F78372A06FF50C80464FC1B4F7B8CEA6056E8E56 from Apt::Source rabbitmq]): $key_source is deprecated and will be removed in the next major release. Please use $source instead.
==> machine11: Notice: Compiled catalog for machine11.puphpet in environment production in 9.64 seconds
==> machine11: Warning: Found multiple default providers for vcsrepo: cvs, git, svn, hg, bzr, dummy; using cvs
==> machine11: Warning: /Apt_key[Add key: 7BD9BF62 from Apt::Source nginx]: The id should be a full fingerprint (40 characters), see README.
==> machine11: Info: Applying configuration version '1464193624'
==> machine11: Notice: /Stage[main]/Puphpet_nginx/File[/var/www]/owner: owner changed 'www-data' to 'root'
==> machine11: Notice: /Stage[main]/Puphpet_mariadb/Exec[Set pidfile parent directory permissions]/returns: executed successfully
==> machine11: Notice: /Stage[main]/Puphpet_server/Exec[dotfiles]/returns: executed successfully
==> machine11: Info: Computing checksum on file /etc/timezone
==> machine11: Info: FileBucket got a duplicate file {md5}4f24b133ba38d8fd565168742c9aedeb
==> machine11: Info: /Stage[main]/Timezone/File[/etc/timezone]: Filebucketed /etc/timezone to puppet with sum 4f24b133ba38d8fd565168742c9aedeb
==> machine11: Notice: /Stage[main]/Timezone/File[/etc/timezone]/content: content changed '{md5}4f24b133ba38d8fd565168742c9aedeb' to '{md5}ae1d80ba85332ae2d9fbbd491b17c043'
==> machine11: Info: /Stage[main]/Timezone/File[/etc/timezone]: Scheduling refresh of Exec[update_timezone]
==> machine11: Notice: /Stage[main]/Timezone/Exec[update_timezone]: Triggered 'refresh' from 1 events
==> machine11: Notice: /Stage[main]/Rabbitmq::Repo::Apt/Apt::Source[rabbitmq]/Apt::Key[Add key: F78372A06FF50C80464FC1B4F7B8CEA6056E8E56 from Apt::Source rabbitmq]/Anchor[apt_key F78372A06FF50C80464FC1B4F7B8CEA6056E8E56 present]: Dependency Apt_key[Add key: F78372A06FF50C80464FC1B4F7B8CEA6056E8E56 from Apt::Source rabbitmq] has failures: true
==> machine11: Error: The id in your manifest F78372A06FF50C80464FC1B4F7B8CEA6056E8E56 and the fingerprint from content/source do not match. Please check there is not an error in the id or check the content/source is legitimate.
==> machine11: Error: /Stage[main]/Rabbitmq::Repo::Apt/Apt::Source[rabbitmq]/Apt::Key[Add key: F78372A06FF50C80464FC1B4F7B8CEA6056E8E56 from Apt::Source rabbitmq]/Apt_key[Add key: F78372A06FF50C80464FC1B4F7B8CEA6056E8E56 from Apt::Source rabbitmq]/ensure: change from absent to present failed: The id in your manifest F78372A06FF50C80464FC1B4F7B8CEA6056E8E56 and the fingerprint from content/source do not match. Please check there is not an error in the id or check the content/source is legitimate.
==> machine11: Warning: /Stage[main]/Rabbitmq::Repo::Apt/Apt::Source[rabbitmq]/Apt::Key[Add key: F78372A06FF50C80464FC1B4F7B8CEA6056E8E56 from Apt::Source rabbitmq]/Anchor[apt_key F78372A06FF50C80464FC1B4F7B8CEA6056E8E56 present]: Skipping because of failed dependencies
==> machine11: Notice: /Stage[main]/Rabbitmq::Repo::Apt/Apt::Source[rabbitmq]/Apt::Setting[list-rabbitmq]/File[/etc/apt/sources.list.d/rabbitmq.list]: Dependency Apt_key[Add key: F78372A06FF50C80464FC1B4F7B8CEA6056E8E56 from Apt::Source rabbitmq] has failures: true
==> machine11: Warning: /Stage[main]/Rabbitmq::Repo::Apt/Apt::Source[rabbitmq]/Apt::Setting[list-rabbitmq]/File[/etc/apt/sources.list.d/rabbitmq.list]: Skipping because of failed dependencies
```

**Comment by dragosprotung** [comment]
I have the same problem.

Temporary solution is to modify `puphpet/puppet/modules/rabbitmq/manifests/install.pp` file by adding `install_options => [ '--force-yes' ],` after `source`

**Comment by lenada** [comment]
It seems rabbitmq.com "updated" their apt "Release Signing Key" 

old
```
pub   1024D/056E8E56 2007-07-06
uid                  RabbitMQ Release Signing Key <info@rabbitmq.com>
sub   2048g/81181462 2007-07-06
```

new

```
pub   4096R/6026DFCA 2016-05-17
uid                  RabbitMQ Release Signing Key <info@rabbitmq.com>
sub   4096R/12EBCE19 2016-05-17
``

**Comment by dragosprotung** [comment]
We need to wait until the fix puppetlabs/puppetlabs-rabbitmq#477 is merged

**Comment by jtreminio** [comment]
Fixed in latest release.
